core/src/main/java/org/arquillian/smart/testing/configuration/ConfigurationLoader.java

```diff
@@ -27,11 +28,24 @@
 
     private static final Logger logger = Log.getLogger();
 
+    /**
+     * Do not use - only for test purposes
+     *
+     * @return A {@link Configuration} instance
+     */
     public static Configuration load() {
         return load(Paths.get("").toAbsolutePath().toFile());
     }
 
     public static Configuration load(File projectDir) {
+        return load(projectDir, null);
```

Comment:
The null is for cases when you don't want to do the lookup https://github.com/arquillian/smart-testing/pull/266/files/813379562e805715ab9e4e87d8449769febdb766#diff-b6b0e49bef8b6669e2f00a5b5658f6ccR45
There could be available `StopCondition.TO_ROOT_DIRECTORY` and also `StopCondition.NO_LOOKUP`
I'm not sure if it is fine to replace the whole `getFirstDirWithConfigOrProjectRootDir` logic by `RecursiveFileLookup.TO_ROOT_DIRECTORY` and similar options because for Maven extension it requires specific information (`MAVEN_PROJECTBASEDIR` environment set) and this is something I didn't want to include into the core module. On the other hand, it's just an env and if I call the prepared instance with the correct name (eg. `TO_MAVEN_PROJECT_ROOT_DIRECTORY`), then it shouldn't be a problem.